lib/board.rb

```diff
@@ -0,0 +1,73 @@
+require 'pry'
+
+class Board
+attr_accessor :cells
+
+def initialize
+  @cells = Array.new(9," ")
+end
+
+def reset!
+  @cells = Array.new(9," ")
+end
+
+
+def display
+  puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+  puts "------------"
+  puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+  puts "-------------"
+  puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+end
+
+
+def position(input)
+  input = input.to_i-1
+  @cells[input]
+end
+
+def full?
+  if @cells.include?(" ")
+    false
+  else
+    true
+  end
+end
+
+
+def turn_count
+  counter = 0
+  @cells.each do |cell|
+    if cell == "X" || cell == "O"
+      counter += 1
+    end
+  end
+    counter
+  end
+
+  def taken?(input)
+    input = input.to_i - 1
+    @cells[input] == "X" || @cells[input] == "O"
+  end
+
+def valid_move?(input)
```

Comment:
You can use the implicit return of the boolean expressions instead of checking for equality to false. 

```
input.to_i.between?(1,9) && !taken(input)
```